src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs

```diff
@@ -495,6 +502,10 @@ private async Task VerifyGeneratedCodeDiagnosticsAsync(ImmutableArray<Diagnostic
             VerifyDiagnosticResults(await GetSortedDiagnosticsAsync(transformedProject, additionalProjects, analyzers, generatedCodeVerifier, cancellationToken).ConfigureAwait(false), analyzers, expectedResults, generatedCodeVerifier);
         }
 
+        /// <summary>
+        /// This test is not related to test a DiagnosticSuppressor, but to test the compiler feature "#pragma warning disable"
+        /// When testing a DiagnosticSuppressor, this test may generate false negatives.
+        /// </summary>
```

Comment:
📝 I'd prefer to focus the documentation on what the message does. This method ensures that if a user places `#pragma warning disable ID` at the top of the file (where `ID` is specific to their diagnostic analyzer), the analyzer diagnostic will be suppressed. The check was designed after a couple of analyzers were identified that always placed a diagnostic on the first line of the file (e.g. the file header validation), making it impossible to place a suppression line _above_ the diagnostic.